vmdb/app/models/mixins/has_many_orchestration_stack_mixin.rb

```diff
@@ -7,8 +7,8 @@ module HasManyOrchestrationStackMixin
              :dependent   => :destroy
 
     has_many :direct_orchestration_stacks,
+             -> { where(:orchestration_stacks => {:ancestry => nil}) },
```

Comment:
I was more asking about the fact that your code doesn't have the extra outer key:  `:orchestration_stack => {:ancestry => nil)`.  But if this code works, then great!